test/unit/clients/connector-client/connector-post-worldpay-check-3ds-flex-config.pact.test.js

```diff
@@ -0,0 +1,127 @@
+const worldpay3dsFlexCredentialsFixtures = require('../../../fixtures/worldpay-3ds-flex-credentials.fixtures')
+const Connector = require('../../../../app/services/clients/connector.client').ConnectorClient
+const PORT = Math.floor(Math.random() * 48127) + 1024
+const connectorClient = new Connector(`http://localhost:${PORT}`)
+const ACCOUNTS_RESOURCE = '/v1/api/accounts'
+const EXISTING_GATEWAY_ACCOUNT_ID = 1
+const CHECK_WORLDPAY_3DS_FLEX_CREDENTIALS = 'worldpay/check-3ds-flex-config'
+
+describe('connector client - check Worldpay 3DS Flex credentials', () => {
+  const provider = new Pact({
+    consumer: 'selfservice',
+    provider: 'connector',
+    port: PORT,
+    log: path.resolve(process.cwd(), 'logs', 'mockserver-integration.log'),
+    dir: path.resolve(process.cwd(), 'pacts'),
+    spec: 2,
+    pactfileWriteMode: 'merge'
+  })
+
+  before(() => provider.setup())
+  afterEach(() => provider.verify())
+  after(() => provider.finalize())
+
+  describe('when a request to check Worldpay 3DS Flex credentials is made', () => {
+    describe('and the credentials are valid and pass the existing format validation', () => {
+      const checkValidWorldpay3dsFlexCredentialsRequest = worldpay3dsFlexCredentialsFixtures.checkValidWorldpay3dsFlexCredentialsRequest()
+      const checkValidWorldpay3dsFlexCredentialsResponse = worldpay3dsFlexCredentialsFixtures.checkValidWorldpay3dsFlexCredentialsResponse()
+      before(done => {
+        const pactifiedRequest = checkValidWorldpay3dsFlexCredentialsRequest.getPactified()
+        const pactifiedResponse = checkValidWorldpay3dsFlexCredentialsResponse.getPactified()
+
+        provider.addInteraction(
+          new PactInteractionBuilder(`${ACCOUNTS_RESOURCE}/${EXISTING_GATEWAY_ACCOUNT_ID}/${CHECK_WORLDPAY_3DS_FLEX_CREDENTIALS}`)
+            .withState('has the credentials')
+            .withUponReceiving('a request to check Worldpay 3DS Flex credentials')
+            .withMethod('POST')
+            .withRequestHeaders({ 'Content-Type': 'application/json' })
+            .withRequestBody(pactifiedRequest.payload)
+            .withStatusCode(200)
+            .withResponseHeaders({ 'Content-Type': 'application/json' })
+            .withResponseBody(pactifiedResponse)
+            .build()
+        ).then(() => done())
+          .catch(done)
+      })
+
+      it('should return valid', done => {
+        const valid3dsFlexCredentials = checkValidWorldpay3dsFlexCredentialsRequest.getPlain()
+        const validResult = checkValidWorldpay3dsFlexCredentialsResponse.getPlain()
+        connectorClient.postCheckWorldpay3dsFlexCredentials(valid3dsFlexCredentials)
+          .should.be.fulfilled.then((response) => {
+            expect(response).to.deep.equal(validResult)
+          }).should.notify(done)
+      })
+    })
+
+    describe('and the credentials are invalid and pass the existing format validation', () => {
```

Comment:
Generally speaking, pact tests are expensive and we don't want to unnecessarily cater for every scenario using a pact test. In the context of this story, the purpose of the pact test is to ensure that `/v1/api/accounts/{accountId}/worldpay/check-3ds-flex-config` does return 
```
{
  "result": "valid"
}
```
and we want to ensure this contract will never be broken.

Tests like this (which check for valid credentials but don't pass the format validation) should be unit/integration tests in pay-connector itself, so this test case can be removed.

Reply:
Thanks for the explanation. I have removed unhappy scenarios.